src/arithmetic/entity_funcs/entity_funcs.c

```diff
@@ -34,6 +35,43 @@ SIValue AR_LABELS(SIValue *argv, int argc) {
 	return SI_ConstStringVal(label);
 }
 
+// returns true if input node contains all specified labels, otherwise false
+SIValue AR_HAS_LABELS(SIValue *argv, int argc) {
+	if(SI_TYPE(argv[0]) == T_NULL) return SI_NullVal();
+
+	bool         res       =  true;
+	Node         *node     =  argv[0].ptrval;
+	SIValue      labels    =  argv[1];
+	EntityID     id        =  ENTITY_GET_ID(node);
+	GraphContext *gc       =  QueryCtx_GetGraphCtx();
+	Graph        *g        =  gc->g;
+
+	// iterate over given labels
+	uint32_t labels_length = SIArray_Length(labels);
+	for (uint32_t i = 0; i < labels_length; i++) {
+		char *label = SIArray_Get(labels, i).stringval;
```

Comment:
Validate that `SIArray_Get(labels, i)` is of type string,
if it isn't see what's the proper way to respond.